### PR TITLE
Prepare release v1.25.0

### PR DESCRIPTION
**Pull request by llorca**
:fireworks: **Highlights**: new `Popover2`/`Tooltip2` components in __labs__, `TagInput` gets __5__ new props, `Table` now supports frozen/sticky columns/rows, new `Icon` component, new `IconName` typings

:book: **Latest docs**: [blueprintjs.com/docs](http://blueprintjs.com/docs)

## General

- 😭 This is my (@giladgray) last release as the Blueprint lead. I'm moving on to other frontend projects inside Palantir, but I'll always be connected to Blueprint (and just an at-mention away), and I may very well be back in a few months! 
  - Core components have been quite stable for a long time so the rest of the team will remain active on `Table` and continue to maintain everything else (though new features are likely to slow down) 
  - I plan to push the Labs components as hard as I can in real-world applications, and we'll continue to improve those components in response to feedback _(yes, that means you have to use the `labs` package if you want it to improve)_
  - But before I go, one final tremendous release to brighten your week:

### 💎  New features
- #1427 🌟  `Icon` component is the CSS shorthand you've always dreamed of:
    ```diff
    private maybeRenderLeftIcon() {		
    -    const { leftIconName } = this.props;		
    -    if (leftIconName == null) {		
    -        return undefined;		
    -    }		
    -    const iconClass = Classes.iconClass(leftIconName);		
    -    return <span className={classNames(Classes.ICON_LARGE, iconClass)} />;		
    +    return <Icon iconName={leftIconName} iconSize={20} />
    }
    ```
    - We've also added typings for `iconName` throughout the library with a new string literal union type, `IconName`, that includes every short and long icon: `add` and `pt-icon-add`. you know what this means: _editor autocomplete_ 😱  icon-autocomplete
    - ⚠️  In introducing this new `IconName` type, we've done our best to ensure compatibility with existing use cases (it supports both short and long names, for instance), but it is likely that `tsc` will complain lightly about a few usages in your codebase
        - If your component accepts `iconName` as a prop to pass to `<Icon />`, you'll need to [change the type](https://github.com/palantir/blueprint/pull/1427/files#diff-4da08b8227d8bcd6f52756f5bad5a2d1L36) from `string` to `IconName`

- #1373 🌟 much improving scrolling support for `Overlay` and centering for `Dialog`
  - We refactored `.pt-overlay` CSS to ensure that it sets the correct `position` at all times, whether inline or scrollable
  - `Dialog` component now renders a `.pt-dialog-container` wrapper as its root element which centers the `.pt-dialog` in a viewport-covering flex container. This provides super reliable centering behavior (#552) and much improved scrolling support (#1313) (though some issues are still present like #1008)
    - ⚠️ If you're rendering the `.pt-dialog` markup by hand, you may want to add the `.pt-dialog-container` parent element for perfect screen centering (and probably remove any custom styles you had); see the CSS API markup in the docs
    - ⚠️ If you're using the `Dialog` component, the new container element will be added automatically and you can likely remove any custom positioning styles you've written

### 🐛  Bug fixes
- #1421 Remove small font-size from small buttons to improve legibility and fix icon centering
small-buttons-centered
- #1434 Move `$output-bourbon-deprecation-warnings` variable to `_variables.scss` to globally disable Bourbon's deprecation warnings in any package that imports Blueprint Sass code

### 🐛  Bug fixes
- #1429 Pass locale to `moment()` for parsing and formatting
- #1424 Speed up the test suite by a full 30 seconds!

## `@blueprintjs/table 1.22.0`
- #1445 🌟 `Table` now supports frozen/sticky columns and rows! Use the `numFrozenColumns` and `numFrozenRows` props and you're set.
29045022-d060c77c-7b76-11e7-806c-054fd2debcfb

### 💎  New features

- #1364 🌟 `Popover2` uses a new positioning engine that resolves our "smart positioning" bugs
popover2-auto
  - [Popper.js](https://popper.js.org) is a massive improvement over [Tether](http://tether.io/) in almost every way!
    - all the fancy flipping behavior you could want _enabled by default_ 
    - endlessly customizable if it isn't perfect _enough_ for you
    - look, it puts the arrow exactly where it's supposed to be. _every time._ what more could a blueprint dev want??
  - All the classic `Popover` features are still supported, with the same names...
  - Except for the handful of Tether-specific props, which are now Popper.js-specific:
    - 🔥 `position: Position` &rarr; `placement: PopperJS.Placement`
    - 🔥 `tetherOptions: ITetherOptions` &rarr; `modifiers: PopperJS.Modifiers`
  - And one special addition:
    - `minimal: boolean` applies minimal styles, which includes removing the arrow and minimizing the transition 
  - This new iteration of the `Popover` API will live in __labs__ until our eventual 2.0 release, when it will be merged into __core__ and will replace Tether entirely
- #1438 🌟 `Tooltip2` exposes the same API as `Tooltip` but is backed by `Popover2` (and has the same API changes noted above)

- #1418, #1419, #1420 🌟  `TagInput` gets a lot of love this week!
tag-input-new-props
  - new `separator` prop allows adding multiple values at once
  - new `onChange` callback prop handles `onAdd` and `onRemove` array manipulation for you. great for simple use cases!
  - new `placeholder` prop disappears when `values` is not empty, like an input placeholder normally does
  - new `leftIconName` and `rightElement` props work exactly like `InputGroup` (with new `IconName` typings!)
  - `values` accepts any valid JSX, from strings to elements to fragments!
    - #1432 falsy values are ignored appropriately, like normal JSX
  - 🔥 __API BREAK__: `onAdd` now receives `string[]` instead of single `string`
    ```diff
    - private handleAdd = (value: string) => this.setState([...this.state.values, value])
    + private handleAdd = (values: string[]) => this.setState([...this.state.values, ...values])
    ```

### 🐛  Bug fixes
- #1403 `QueryList` `onKeyDown` is invoked for _all_ keys (🎩  @paulpooch)
- #1422 `MultiSelect` placeholder can be overridden through `inputProps.placeholder` (🎩  @sushain97)

### CODE REVIEW

**Comment by giladgray** [comment]
@llorca please update to include latest master, I just merged #1432 (added to release notes in description)

File: packages/labs/package.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-  "version": "0.5.0",
+  "version": "0.6.0",
   "description": "Incubator for unstable and in-development Blueprint components",
   "main": "dist/index.js",
   "typings": "dist/index.d.ts",
   "style": "dist/blueprint-labs.css",
   "unpkg": "dist/labs.bundle.js",
   "dependencies": {
-    "@blueprintjs/core": "^1.23.0",
+    "@blueprintjs/core": "^1.25.0",
```
Comment (giladgray) [inline packages/labs/package.json]: this is correct because of `Icon` in `TagInput`

File: packages/table/package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "watch": "onchange 'src/**' 'preview/*.ts*' -- npm-run-all build:gulp build:preview"
   },
   "dependencies": {
-    "@blueprintjs/core": "^1.23.0",
+    "@blueprintjs/core": "^1.25.0",
```
Comment (giladgray) [inline packages/table/package.json]: revert, table does not need latest changes

**Comment by cmslewis** [comment]
Delete the incorrect/duplicate bugfixes section in packages/table.

**Comment by cmslewis** [comment]
(FYI just standardized some capitalizations.)

**Comment by adidahiya** [comment]
Is this intended? the Tooltip2 button gets disabled when I click on it:

image

**Comment by giladgray** [comment]
@adidahiya huge diffs are sensible for a few reasons:

1. i moved `_icons.scss` deeper in the tree, which reordered the rendered CSS. note how `.pt-icon-*` appears much lower in the .css file, instead of right at the top
2. the numbers are large, but the difference btw additions and deletions is small because it's mostly _movement_ (see point 1)
3. the built .js files are much larger cuz there's a lot of new code and a new external dep (popper.js)

time to ship! 🚢

**Pull request merged**